allennlp/models/encoder_decoders/wikitables_semantic_parser.py

```diff
@@ -271,6 +310,166 @@ def forward(self,  # type: ignore
             # TODO(matt): compute accuracy here.
             return outputs
 
+    @staticmethod
+    def _get_neighbor_indexes(worlds: List[WikiTablesWorld],
+                              num_entities: int,
+                              tensor: torch.Tensor) -> Variable:
+        """
+        This method returns the indices of each entity's neighbors. A tensor
+        is accepted as a parameter for copying purposes
+
+        Parameters
+        ----------
+        worlds : ``List[WikiTablesWorld]``
+        num_entities : ``int``
+        tensor : ``torch.Tensor``
+            Used for copying the constructed list onto the right device.
+
+        Returns
+        -------
+        A ``torch.autograd.Variable`` with shape ``(batch_size, num_entities, num_entities)``.
```

Comment:
Does this need to be `num_entities`?  Could it just be `num_neighbors`, where you find the max num neighbors in this method, and only pad to that amount?  You'd save quite a bit of memory, and computation, I think, because the amount of padding would be a whole lot less.